pkg/controller/controller.go

```diff
@@ -1145,6 +1145,14 @@ func (p *csiProvisioner) Delete(ctx context.Context, volume *v1.PersistentVolume
 	storageClassName := util.GetPersistentVolumeClass(volume)
 	if len(storageClassName) != 0 {
 		if storageClass, err := p.scLister.Get(storageClassName); err == nil {
+			if migratedVolume && storageClass.Provisioner == p.supportsMigrationFromInTreePluginName {
```

Comment:
Is it possible to add a unit test to https://github.com/kubernetes-csi/external-provisioner/blob/master/pkg/controller/controller_test.go#L4310 to cover this scenario?

Reply:
I'm not familiar with gomock new, but I will try to do it.